src/app/components/annotator/text-annotator/text-annotator.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { AnnotatorColors, AppConfig } from 'src/app/app.config';
+import { AnnotatorService } from 'src/app/services/annotator/annotator.service';
+
+interface AnnotatorPosition {
+  x: number;
+  y: number;
+}
+@Component({
+  selector: 'evt-text-annotator',
+  templateUrl: './text-annotator.component.html',
+  styleUrls: ['./text-annotator.component.scss'],
+})
+export class TextAnnotatorComponent implements OnDestroy {
+  private subscriptions: Subscription[] = [];
+  public highlightcolors: AnnotatorColors = AppConfig.evtSettings.edition.annotatorColors;
+  public selectedText: string;
+  public showAdder = false;
+  public menuPosition: AnnotatorPosition = { x: 0, y: 0 };
+  public annoType: string;
+  public updateMode = true;
+
+  constructor(
+    private annotator: AnnotatorService,
+  ) {
+    this.subscriptions.push(this.annotator.textSelection$
+      .subscribe((selection) => {
+        this.selectedText = selection.toString();
+        if (/\S/.test(selection.toString())){
```

Comment:
since you have already computed `selection.toString()` you can reuse `this.selectedText` inside `test`